File: algorithms/05_csharp/dp/kadane.cs
using System;

namespace Algorithms.DynamicProgramming
{
    public static class Kadane
    {
        public static int MaxSubarraySum(int[] array)
        {
            int best = array[0], current = array[0];
            for (int i = 1; i < array.Length; i++)
            {
                current = Math.Max(array[i], current + array[i]);
                best = Math.Max(best, current);
            }
            return best;
        }

        public static void Main()
        {
            int[] a = { -2, 1, -3, 4, -1, 2, 1, -5, 4 };
            bool ok = MaxSubarraySum(a) == 6;
            Console.WriteLine(ok ? "[C# Kadane] ... verified" : "[C# Kadane] ... FAILED");
        }
    }
}

File: algorithms/06_javascript/dp/kadane.js
function maxSubarraySum(arr) {
  let best = -Infinity, cur = 0;
  for (const x of arr) {
    cur = Math.max(x, cur + x);
    best = Math.max(best, cur);
  }
  return best;
}

module.exports = { maxSubarraySum };

if (require.main === module) {
  if (maxSubarraySum([-2, 1, -3, 4, -1, 2, 1, -5, 4]) !== 6) throw new Error("kadane failed");
  if (maxSubarraySum([-1, -2, -3]) !== -1) throw new Error("all-negative case failed");
  console.log("[JavaScript Kadane] Maximum subarray sum verified");
}
